### ISSUE TITLE
Chen_EffortEstimationMetrics

### ISSUE THREAD

**Issue by obaysal** [researchart/icsme16#14]

**Comment by ctreude** [comment]
The authors have conducted a systematic literature to review the state of the art of maintenance effort estimation approaches for open source software. The final set of papers that they investigated contains 29 papers, and they extracted a total of 85 metrics from this data. In addition to the titles of the papers and the detailed descriptions of the metrics along with a matrix that maps metrics to papers, the artifact also contains information on the estimation accuracy of different approaches, based on the subset of 20 papers that include the corresponding data.

While certainly insightful and useful, the authors could have put more effort into making the artifact usable. In the current version, information is not really interlinked, e.g., the links to the 29 papers (or even the author names) are missing, and the metrics tables don't provide any traceability to the papers either. A user has to manually go back and forth between the different sites to connect information. All data is only available as HTML which makes further processing unnecessarily difficult. 

In summary, the artifact seems more like an online appendix for the paper that includes data omitted from the paper for space reasons. More work would be needed to turn it into an easily-usable research artifact.

The pdf requires some proof-reading for grammatical mistakes.

**Comment by Felienne** [comment]
This artefact combines the data of open source metrics of 29 papers, gathering a total of 85 metrics. I agree with @ctreude that is idea is interesting, there are some improvements to be suggested:
- Link to (open, preprint) versions of the paper, or rather include the PDFs in the dataset to allow people to find it.  
- Save the data in a better format (JSON? csv?)

A few minor issues:

For researchers who devote to build the quality models of open source software, -> who devote their time or effort?
Open source software are -> singular is more common I think

29 careful selected studies -> The selection mechanisms are probably in the full paper, but I would summarize here briefly.

**Comment by timm** [comment]
@Felienne and @ctreude 
please comment on "?maybe accept" or "?maybe reject"

**Comment by ctreude** [comment]
@timm Ultimately, the artifact isn't nearly as useful as it could be, so I'm going for "?maybe reject".

**Comment by Felienne** [comment]
Agreed.

**Comment by davidlo83** [comment]
Dear Hong, Lin, Celia, Qing, Barry, Tim, and Olga,

I've read the 2-page description of the artifact and visit the webpage containing the artifact. The following is my review:

[Insightful]

> Timely (i.e., addresses a problem that is most current and most pressing)?
> A large number of research has been done on maintenance effort estimation. It is hard for researchers and practitioners to read all those studies. This work compiles a list of metrics used to estimate maintenance effort for open source projects from many past studies, which is timely.
> 
> Makes researchers “smarter” in some way (e.g., identifies and fills some significant gap in prior work)?
> The list can help researchers to get started faster in the area of software maintenance effort estimation. Since the work is a systematic literature review, no new content (i.e., metric) is introduced, which is okay.

[Useful]

> Serves a useful purpose?
> The list provides a "cheat sheet"-like resource to researchers and practitioners which is good.  
> 
> Serves a purpose that would otherwise be tedious, prolonged, awkward, or impossible?
> Without the list, researchers and practitioners need to read a large number of papers, which is tedious.
> 
> Cost-effective?
> N/A. The artifact is not a software system and thus cost-effectiveness (i.e., efficiency) is not a relevant point.

[Usable (co-reviewed by Xuan-Bach D. Le)]
- The artefact says: "The list of the 29 studies can be found on: http://goo.gl/gUii20." However, we cannot find it anywhere...
- 85 metrics are provided, but there were no explanation for the meaning of the columns in the 85 presented metrics. It is hard to guess what they mean ... Furthermore, it lacks of a tutorial demonstrating examples on how users can make use of the data.
- Overall, the authors put some effort to make to data publicly available, but still, it is not easy to understand the organization of the data at the first place.

[Recommendation]

maybe accept?, after some modifications (e.g., by including more information on the webpage -- http://goo.gl/gUii20) to make the artifact more understandable and easily usable by others.

Thank you and have a great week !

David

**Comment by davidlo83** [comment]
Dear Christoph, Felienne, Tim, and Olga,

Maybe we can allow the authors to improve the understandability of the artifact and consider for acceptance :-) ? I think they have all the information and just need to put them together. Just my 2 cents :-)

Please kindly advise. Thank you !

Yours Sincerely,

David

**Comment by ctreude** [comment]
I won't stand in the way of an acceptance here, but did base my recommendation on what's there at the moment rather than the potential of an improved version.

@timm and @obaysal, what's the protocol here?

**Comment by timm** [comment]
@davidlo83 : your own review comments on drawbacks with the artifact. the other reviewers would need to see those changes BEFORE they could accept it. but i am worried that if we are ask the authors to make those changes then that would be most unfair to other authors in this track.

so i would lean towards ?maybe reject.

@obaysal : opinion?

**Comment by davidlo83** [comment]
Dear all,

I'm okay with the decision of the majority :-) Some other artifacts were modified during the reviewing process though and thus I thought we could extend this opportunity too to the authors of this artifact :-) I understand that this may be too late though ... 

Please kindly advise. Thank you and have a great week ahead !

Yours Sincerely,

David

P.S. Sorry for the late reply since I'm currently travelling ...

**Comment by obaysal** [comment]
Based on the reviews, I am leaning towards a reject here. The dataset seems to be too small for conducting a systematic study.
